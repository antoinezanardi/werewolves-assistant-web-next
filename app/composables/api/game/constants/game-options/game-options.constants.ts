import type { GameOptions } from "~/composables/api/game/types/game-options/game-options.class";

const DEFAULT_GAME_OPTIONS: GameOptions = {
  composition: { isHidden: false },
  votes: { canBeSkipped: true },
  roles: {
    areRevealedOnDeath: true,
    doSkipCallIfNoTarget: false,
    sheriff: {
      isEnabled: true,
      electedAt: {
        turn: 1,
        phaseName: "night",
      },
      hasDoubledVote: true,
      mustSettleTieInVotes: true,
    },
    bigBadWolf: { isPowerlessIfWerewolfDies: true },
    whiteWerewolf: { wakingUpInterval: 2 },
    seer: {
      isTalkative: true,
      canSeeRoles: true,
    },
    cupid: {
      lovers: { doRevealRoleToEachOther: false },
      mustWinWithLovers: false,
    },
    littleGirl: { isProtectedByDefender: false },
    defender: { canProtectTwice: false },
    elder: {
      livesCountAgainstWerewolves: 2,
      doesTakeHisRevenge: true,
    },
    idiot: { doesDieOnElderDeath: true },
    twoSisters: { wakingUpInterval: 2 },
    threeBrothers: { wakingUpInterval: 2 },
    fox: { isPowerlessIfMissesWerewolf: true },
    bearTamer: { doesGrowlOnWerewolvesSide: true },
    stutteringJudge: { voteRequestsCount: 1 },
    wildChild: { isTransformationRevealed: false },
    wolfHound: {
      isChosenSideRevealed: false,
      isSideRandomlyChosen: false,
    },
    thief: {
      mustChooseBetweenWerewolves: true,
      isChosenCardRevealed: false,
      additionalCardsCount: 2,
    },
    piedPiper: {
      charmedPeopleCountPerNight: 2,
      isPowerlessOnWerewolvesSide: true,
    },
    scandalmonger: { markPenalty: 2 },
    witch: { doesKnowWerewolvesTargets: true },
    prejudicedManipulator: { isPowerlessOnWerewolvesSide: true },
    actor: {
      isPowerlessOnWerewolvesSide: true,
      additionalCardsCount: 3,
    },
  },
} as const;

export { DEFAULT_GAME_OPTIONS };