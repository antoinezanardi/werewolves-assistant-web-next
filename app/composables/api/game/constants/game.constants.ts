import { PLAYER_GROUPS } from "~/composables/api/game/constants/player/player.constants";
import type { PlayerAttributeName } from "~/composables/api/game/types/players/player-attribute/player-attribute.types";
import type { PlayerGroup } from "~/composables/api/game/types/players/player.types";
import { ROLE_NAMES } from "~/composables/api/role/constants/role.constants";
import type { RoleName } from "~/composables/api/role/types/role.types";

const GAME_STATUSES = [
  "playing",
  "over",
  "canceled",
] as const;

const GAME_SOURCES = [
  ...PLAYER_GROUPS,
  ...Object.values(ROLE_NAMES),
  "sheriff",
] as const satisfies readonly (PlayerAttributeName | PlayerGroup | RoleName)[];

const MIN_PLAYERS_IN_GAME = 4;

const MAX_PLAYERS_IN_GAME = 40;

const MIN_PLAYERS_IN_GROUP = 2;

export {
  GAME_STATUSES,
  GAME_SOURCES,
  MIN_PLAYERS_IN_GAME,
  MAX_PLAYERS_IN_GAME,
  MIN_PLAYERS_IN_GROUP,
};