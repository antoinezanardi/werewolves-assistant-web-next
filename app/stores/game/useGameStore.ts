import type { AsyncDataRequestStatus } from "nuxt/app";
import { defineStore } from "pinia";
import { unique } from "radash";

import type { MakeGamePlayDto } from "~/composables/api/game/dto/make-game-play/make-game-play.dto";
import type { GameOptions } from "~/composables/api/game/types/game-options/game-options.class";
import { Game } from "~/composables/api/game/types/game.class";
import { useFetchGames } from "~/composables/api/game/useFetchGames";
import { StoreIds } from "~/stores/enums/store.enum";
import { useGameEventsStore } from "~/stores/game/game-event/useGameEventsStore";

const useGameStore = defineStore(StoreIds.GAME, () => {
  const {
    getGame: fetchGameFromApi,
    cancelGame: cancelGameFromApi,
    makeGamePlay: makeGamePlayFromApi,
  } = useFetchGames();

  const game = ref<Game>(new Game());
  const gameOptions = computed<GameOptions>(() => game.value.options);

  const gamePlayerGroups = computed<string[]>(() => unique(game.value.players.map(player => player.group)).filter(Boolean) as string[]);

  const fetchingGameStatus = ref<AsyncDataRequestStatus>("idle");
  const cancelingGameStatus = ref<AsyncDataRequestStatus>("idle");
  const makingGamePlayStatus = ref<AsyncDataRequestStatus>("idle");

  const { resetGameEventIndex } = useGameEventsStore();

  function resetGame(): void {
    game.value = new Game();
  }

  async function fetchAndSetGame(gameId: string): Promise<void> {
    fetchingGameStatus.value = "pending";
    const fetchedGame = await fetchGameFromApi(gameId);
    if (!fetchedGame) {
      fetchingGameStatus.value = "error";

      return;
    }
    resetGameEventIndex();
    game.value = fetchedGame;
    fetchingGameStatus.value = "success";
  }

  async function cancelGame(): Promise<void> {
    cancelingGameStatus.value = "pending";
    const canceledGame = await cancelGameFromApi(game.value._id);
    if (!canceledGame) {
      fetchingGameStatus.value = "error";

      return;
    }
    game.value = canceledGame;
    cancelingGameStatus.value = "success";
  }

  async function makeGamePlay(makeGamePlayDto: MakeGamePlayDto): Promise<void> {
    makingGamePlayStatus.value = "pending";
    const playedGame = await makeGamePlayFromApi(game.value._id, makeGamePlayDto);
    if (!playedGame) {
      fetchingGameStatus.value = "error";

      return;
    }
    resetGameEventIndex();
    game.value = playedGame;
    makingGamePlayStatus.value = "success";
  }

  async function skipGamePlay(): Promise<void> {
    return makeGamePlay({});
  }
  return {
    game,
    gameOptions,
    gamePlayerGroups,
    fetchingGameStatus,
    cancelingGameStatus,
    makingGamePlayStatus,
    resetGame,
    fetchAndSetGame,
    cancelGame,
    makeGamePlay,
    skipGamePlay,
  };
});

export { useGameStore };