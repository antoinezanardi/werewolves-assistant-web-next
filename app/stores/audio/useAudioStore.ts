import { useLocalStorage } from "@vueuse/core";
import { Howl, Howler } from "howler";
import { draw } from "radash";
import { defineStore } from "pinia";

import type { GamePhaseName } from "~/composables/api/game/types/game-phase/game-phase.types";
import { BACKGROUND_AUDIO_NAMES, DEFAULT_AUDIO_SETTINGS, SOUND_EFFECT_NAMES } from "~/stores/audio/constants/audio.constants";
import type { AudioSettings, BackgroundAudioName, SoundEffectName } from "~/stores/audio/types/audio.types";
import { StoreIds } from "~/stores/enums/store.enum";
import { LocalStorageKeys } from "~/utils/enums/local-storage.enums";

const useAudioStore = defineStore(StoreIds.AUDIO, () => {
  const audioSettingsFromLocalStorage = useLocalStorage<AudioSettings>(LocalStorageKeys.AUDIO_SETTINGS, DEFAULT_AUDIO_SETTINGS, { mergeDefaults: true });

  const isMuted = ref<boolean>(audioSettingsFromLocalStorage.value.isMuted);

  const soundEffects = Object.fromEntries(SOUND_EFFECT_NAMES.map(name => [name, createSoundEffect(name)])) as Record<SoundEffectName, Howl>;

  const backgroundAudios = Object.fromEntries(BACKGROUND_AUDIO_NAMES.map(name => [name, createBackgroundAudio(name)])) as Record<BackgroundAudioName, Howl>;

  const playingBackgroundAudioName = ref<BackgroundAudioName>();

  const nightBackgroundAudioNames = Object.keys(backgroundAudios).filter((name): name is BackgroundAudioName => name.startsWith("night-"));

  const dayBackgroundAudioNames = Object.keys(backgroundAudios).filter((name): name is BackgroundAudioName => name.startsWith("day-"));

  function createSoundEffect(src: SoundEffectName): Howl {
    return new Howl({
      preload: false,
      src: [`/audio/sound-effects/${src}.webm`],
    });
  }

  function createBackgroundAudio(src: BackgroundAudioName): Howl {
    return new Howl({
      preload: false,
      src: [`/audio/audio-backgrounds/${src}.webm`],
      loop: true,
    });
  }

  function setHowlerAudioSettingsFromAudioStoreState(): void {
    Howler.mute(isMuted.value);
  }

  function loadSoundEffects(): void {
    Object.values(soundEffects).forEach(soundEffect => soundEffect.load());
  }

  function loadBackgroundAudios(): void {
    Object.values(backgroundAudios).forEach(backgroundAudio => backgroundAudio.load());
  }

  function loadAllAudios(): void {
    loadBackgroundAudios();
    loadSoundEffects();
  }

  function playSoundEffect(soundEffectName: SoundEffectName): void {
    soundEffects[soundEffectName].play();
  }

  function fadeOutPlayingBackgroundAudio(): void {
    if (!playingBackgroundAudioName.value) {
      return;
    }
    const fadeOutDuration = 1000;
    const playingBackgroundAudio = backgroundAudios[playingBackgroundAudioName.value];
    playingBackgroundAudio.fade(1, 0, fadeOutDuration);
    playingBackgroundAudioName.value = undefined;
    setTimeout(() => playingBackgroundAudio.stop(), fadeOutDuration);
  }

  function playBackgroundAudio(backgroundAudioName: BackgroundAudioName): void {
    const fadeInDuration = 1000;
    backgroundAudios[backgroundAudioName].fade(0, 1, fadeInDuration);
    backgroundAudios[backgroundAudioName].play();
    playingBackgroundAudioName.value = backgroundAudioName;
  }

  function playRandomGamePhaseBackgroundAudio(gamePhase: GamePhaseName): void {
    const backgroundAudioNames = gamePhase === "night" ? nightBackgroundAudioNames : dayBackgroundAudioNames;
    if (playingBackgroundAudioName.value && backgroundAudioNames.includes(playingBackgroundAudioName.value)) {
      return;
    }
    const randomGamePhaseBackgroundAudioName = draw(backgroundAudioNames);
    if (!randomGamePhaseBackgroundAudioName) {
      return;
    }
    fadeOutPlayingBackgroundAudio();
    playBackgroundAudio(randomGamePhaseBackgroundAudioName);
  }

  function setMute(isAudioMuted: boolean): void {
    isMuted.value = isAudioMuted;
    Howler.mute(isAudioMuted);
    audioSettingsFromLocalStorage.value.isMuted = isAudioMuted;
  }

  function toggleMute(): void {
    setMute(!isMuted.value);
  }
  return {
    audioSettingsFromLocalStorage,
    isMuted,
    soundEffects,
    backgroundAudios,
    playingBackgroundAudioName,
    nightBackgroundAudioNames,
    dayBackgroundAudioNames,
    setHowlerAudioSettingsFromAudioStoreState,
    loadSoundEffects,
    loadBackgroundAudios,
    loadAllAudios,
    playSoundEffect,
    fadeOutPlayingBackgroundAudio,
    playBackgroundAudio,
    playRandomGamePhaseBackgroundAudio,
    setMute,
    toggleMute,
  };
});

export { useAudioStore };