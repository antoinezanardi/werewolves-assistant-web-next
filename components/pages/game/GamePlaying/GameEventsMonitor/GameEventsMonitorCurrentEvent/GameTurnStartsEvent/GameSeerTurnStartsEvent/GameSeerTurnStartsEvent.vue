<template>
  <GameEventWithTexts
    id="game-seer-turn-starts-event"
    :texts="gameEventTexts"
  >
    <GameEventFlippingPlaySourcePlayersCard/>
  </GameEventWithTexts>
</template>

<script setup lang="ts">
import { storeToRefs } from "pinia";
import GameEventFlippingPlaySourcePlayersCard from "~/components/shared/game/game-event/GameEventFlippingPlayerCard/GameEventFlippingPlaySourcePlayersCard/GameEventFlippingPlaySourcePlayersCard.vue";
import GameEventWithTexts from "~/components/shared/game/game-event/GameEventWithTexts/GameEventWithTexts.vue";
import { useGameStore } from "~/stores/game/useGameStore";

const gameStore = useGameStore();
const { game } = storeToRefs(gameStore);

const { t } = useI18n();

const canSeerSeeRoles = computed<boolean>(() => game.value.options.roles.seer.canSeeRoles);

const gameEventTexts = computed<string[]>(() => {
  if (canSeerSeeRoles.value) {
    return [t("components.GameSeerTurnStartsEvent.seerSeesRoles")];
  }
  return [t("components.GameSeerTurnStartsEvent.seerSeesSides")];
});
</script>