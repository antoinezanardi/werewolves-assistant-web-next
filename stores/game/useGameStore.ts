import type { AsyncDataRequestStatus } from "#app/composables/asyncData";
import { defineStore } from "pinia";

import { Game } from "~/composables/api/game/types/game.class";
import { useFetchGames } from "~/composables/api/game/useFetchGames";
import { StoreIds } from "~/stores/enums/store.enum";

const useGameStore = defineStore(StoreIds.GAME, () => {
  const { getGame: fetchGameFromApi } = useFetchGames();

  const game = ref<Game>(new Game());
  const fetchingGameStatus = ref<AsyncDataRequestStatus>("idle");

  async function fetchAndSetGame(gameId: string): Promise<void> {
    fetchingGameStatus.value = "pending";
    const fetchedGame = await fetchGameFromApi(gameId);
    if (!fetchedGame) {
      fetchingGameStatus.value = "error";

      return;
    }
    game.value = fetchedGame;
    fetchingGameStatus.value = "success";
  }
  return {
    game,
    fetchingGameStatus,
    fetchAndSetGame,
  };
});

export { useGameStore };