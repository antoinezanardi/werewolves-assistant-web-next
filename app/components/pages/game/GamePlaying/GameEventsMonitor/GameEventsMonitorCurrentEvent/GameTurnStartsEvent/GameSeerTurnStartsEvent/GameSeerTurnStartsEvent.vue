<template>
  <GameEventWithTexts
    id="game-seer-turn-starts-event"
    :texts="gameEventTexts"
  >
    <GameEventFlippingPlaySourcePlayersCard/>
  </GameEventWithTexts>
</template>

<script setup lang="ts">
import { storeToRefs } from "pinia";

import GameEventFlippingPlaySourcePlayersCard from "~/components/shared/game/game-event/GameEventFlippingPlayerCard/GameEventFlippingPlaySourcePlayersCard/GameEventFlippingPlaySourcePlayersCard.vue";
import GameEventWithTexts from "~/components/shared/game/game-event/GameEventWithTexts/GameEventWithTexts.vue";
import { useAudioStore } from "~/stores/audio/useAudioStore";
import { useGameStore } from "~/stores/game/useGameStore";

const gameStore = useGameStore();
const { game } = storeToRefs(gameStore);

const audioStore = useAudioStore();
const { playSoundEffect } = audioStore;

const { t } = useI18n();

const canSeerSeeRoles = computed<boolean>(() => game.value.options.roles.seer.canSeeRoles);

const gameEventTexts = computed<string[]>(() => {
  if (canSeerSeeRoles.value) {
    return [t("components.GameSeerTurnStartsEvent.seerSeesRoles")];
  }
  return [t("components.GameSeerTurnStartsEvent.seerSeesSides")];
});

playSoundEffect("mystical-mood");
</script>