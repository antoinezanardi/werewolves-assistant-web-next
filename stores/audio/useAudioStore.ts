import { Howl } from "howler";
import { defineStore } from "pinia";
import { BACKGROUND_AUDIO_NAMES, SOUND_EFFECT_NAMES } from "~/stores/audio/constants/audio.constants";
import type { BackgroundAudioName, SoundEffectName } from "~/stores/audio/types/audio.types";
import { StoreIds } from "~/stores/enums/store.enum";

const useAudioStore = defineStore(StoreIds.AUDIO, () => {
  const soundEffects = Object.fromEntries(SOUND_EFFECT_NAMES.map(name => [name, createSoundEffect(name)])) as Record<SoundEffectName, Howl>;

  const backgroundAudios = Object.fromEntries(BACKGROUND_AUDIO_NAMES.map(name => [name, createBackgroundAudio(name)])) as Record<BackgroundAudioName, Howl>;

  function createSoundEffect(src: SoundEffectName): Howl {
    return new Howl({
      preload: false,
      src: [`/audio/sound-effects/${src}.webm`],
    });
  }

  function createBackgroundAudio(src: BackgroundAudioName): Howl {
    return new Howl({
      preload: false,
      src: [`/audio/audio-backgrounds/${src}.webm`],
      loop: true,
      html5: true,
    });
  }

  function loadSoundEffects(): void {
    Object.values(soundEffects).forEach(soundEffect => soundEffect.load());
  }

  function loadBackgroundAudios(): void {
    Object.values(backgroundAudios).forEach(backgroundAudio => backgroundAudio.load());
  }

  function loadAllAudios(): void {
    loadBackgroundAudios();
    loadSoundEffects();
  }

  function playSoundEffect(soundEffectName: SoundEffectName): void {
    soundEffects[soundEffectName].play();
  }

  function playBackgroundAudio(backgroundAudioName: BackgroundAudioName): void {
    backgroundAudios[backgroundAudioName].play();
  }
  return {
    soundEffects,
    backgroundAudios,
    loadSoundEffects,
    loadBackgroundAudios,
    loadAllAudios,
    playSoundEffect,
    playBackgroundAudio,
  };
});

export { useAudioStore };