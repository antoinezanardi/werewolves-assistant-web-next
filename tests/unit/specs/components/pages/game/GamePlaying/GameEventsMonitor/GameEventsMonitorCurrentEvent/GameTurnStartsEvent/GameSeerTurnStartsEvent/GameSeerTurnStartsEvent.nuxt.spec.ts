import { createTestingPinia } from "@pinia/testing";
import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import GameSeerTurnStartsEvent from "~/components/pages/game/GamePlaying/GameEventsMonitor/GameEventsMonitorCurrentEvent/GameTurnStartsEvent/GameSeerTurnStartsEvent/GameSeerTurnStartsEvent.vue";
import { DEFAULT_GAME_OPTIONS } from "~/composables/api/game/constants/game-options/game-options.constants";
import { useAudioStore } from "~/stores/audio/useAudioStore";
import { StoreIds } from "~/stores/enums/store.enum";
import { useGameStore } from "~/stores/game/useGameStore";
import { createFakeGame } from "@tests/unit/utils/factories/composables/api/game/game.factory";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

describe("Game Seer Turn Starts Event Component", () => {
  let wrapper: ReturnType<typeof mount<typeof GameSeerTurnStartsEvent>>;
  const testingPinia = { initialState: { [StoreIds.GAME]: { game: createFakeGame({ options: DEFAULT_GAME_OPTIONS }) } } };

  async function mountGameSeerTurnStartsEventComponent(options: ComponentMountingOptions<typeof GameSeerTurnStartsEvent> = {}):
  Promise<ReturnType<typeof mount<typeof GameSeerTurnStartsEvent>>> {
    return mountSuspendedComponent(GameSeerTurnStartsEvent, {
      global: { plugins: [createTestingPinia(testingPinia)] },
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountGameSeerTurnStartsEventComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should match snapshot when rendered without shallow rendering.", async() => {
    wrapper = await mountGameSeerTurnStartsEventComponent({ shallow: false });

    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should play mystical mood sound effect when rendered.", () => {
    const audioStore = useAudioStore();

    expect(audioStore.playSoundEffect).toHaveBeenCalledExactlyOnceWith("mystical-mood");
  });

  describe("Game Event Texts", () => {
    it("should pass event texts when rendered.", () => {
      const gameEventWithTextsComponent = wrapper.findComponent<typeof GameSeerTurnStartsEvent>("#game-seer-turn-starts-event");
      const expectedTexts: string[] = ["components.GameSeerTurnStartsEvent.seerSeesRoles"];
      const expectedTextsAsString = expectedTexts.join(",");

      expect(gameEventWithTextsComponent.attributes("texts")).toBe(expectedTextsAsString);
    });

    it("should pass event texts with seer can see side of the player when the option is activated.", async() => {
      const gameStore = useGameStore();
      gameStore.game.options.roles.seer.canSeeRoles = false;
      await nextTick();
      const gameEventWithTextsComponent = wrapper.findComponent<typeof GameSeerTurnStartsEvent>("#game-seer-turn-starts-event");
      const expectedTexts: string[] = ["components.GameSeerTurnStartsEvent.seerSeesSides"];
      const expectedTextsAsString = expectedTexts.join(",");

      expect(gameEventWithTextsComponent.attributes("texts")).toBe(expectedTextsAsString);
    });
  });
});