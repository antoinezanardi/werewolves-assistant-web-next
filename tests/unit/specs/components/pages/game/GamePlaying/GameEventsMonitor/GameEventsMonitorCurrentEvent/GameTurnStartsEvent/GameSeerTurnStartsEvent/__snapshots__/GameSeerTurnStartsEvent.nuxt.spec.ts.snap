// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Seer Turn Starts Event Component > should match snapshot when rendered without shallow rendering. 1`] = `
"<div id="game-seer-turn-starts-event" class="flex flex-col h-full">
  <div id="game-event-default-slot-container" class="grow">
    <div class="flex h-full items-center justify-center">
      <div id="game-event-flipping-player-card" class="flex flex-col items-center">
        <div class="flex-none" style="perspective: 1000; width: 200px; height: 200px;" id="game-event-flipping-player-role">
          <div class="flipper" style="transition: 0.6s; transform-style: preserve-3d; position: relative; width: 100%; height: 100%; transition-duration: 0.5s;">
            <div class="front" style="transform-style: preserve-3d; backface-visibility: hidden; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; z-index: 2; transform: rotateY(0deg);"><img width="200" height="200" alt="components.RoleImage.back" data-nuxt-img="" class="border-4 border-gray-800 role-image rounded-lg h-[200pxpx] w-[200pxpx]" id="front-role-image" src="/svg/misc/infinite-spinner.svg">
              <!--v-if-->
            </div>
            <div class="back" style="transform-style: preserve-3d; backface-visibility: hidden; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; transform: rotateY(180deg);"><img width="200" height="200" alt="components.RoleImage.back" data-nuxt-img="" class="border-4 border-gray-800 role-image rounded-lg h-[200pxpx] w-[200pxpx]" id="back-role-image" src="/svg/misc/infinite-spinner.svg">
              <!--v-if-->
            </div>
          </div>
        </div>
        <transition-stub mode="out-in" name="fade" appear="false" persisted="false" css="true">
          <!--v-if-->
        </transition-stub>
      </div>
    </div>
  </div>
  <div id="game-event-texts-manager" class="flex gap-4 h-1/2 items-center"><button id="previous-event-text-button" aria-label="Back to the previous event text" class="ps-4 text-gray-500" disabled="" type="button"><i class="fa fa-3x fa-chevron-left"></i></button>
    <div id="game-event-text-container" class="flex grow h-1/2 items-center justify-center">
      <transition-stub mode="out-in" name="fade" appear="false" persisted="false" css="true">
        <p id="current-event-text" class="!mb-0 !text-2xl text-center">components.GameSeerTurnStartsEvent.seerSeesRoles</p>
      </transition-stub>
    </div><button id="next-event-text-button" aria-label="Next event text" class="pe-4" type="button" data-pd-tooltip="true"><i class="fa fa-3x fa-chevron-right"></i></button>
  </div>
</div>"
`;

exports[`Game Seer Turn Starts Event Component > should match snapshot when rendered. 1`] = `"<game-event-with-texts-stub texts="components.GameSeerTurnStartsEvent.seerSeesRoles" id="game-seer-turn-starts-event"></game-event-with-texts-stub>"`;
